File: app/src/main/java/com/kfang/fencemelibrary/presentation/SharedPreferencesRepository.java
package com.kfang.fencemelibrary.presentation;

/**
 * Interface used for Shared Preferences
 **/

interface SharedPreferencesRepository {
    boolean stayAwakeDuringTimer();

    int pointsToWin();

    boolean popupOnScoreChange();

    boolean restoreOnAppReset();

    boolean vibrateOnTimerFinish();

    boolean enableDoubleTouch();

    boolean vibrateOnTimerToggle();

    boolean pauseOnScoreChange();

    int getBoutLengthMinutes();

}

File: app/src/main/java/com/kfang/fencemelibrary/presentation/SharedPreferencesRepository.kt
package com.kfang.fencemelibrary.presentation

/**
 * Interface used for Shared Preferences
 */

internal interface SharedPreferencesRepository {
    fun stayAwakeDuringTimer(): Boolean

    fun pointsToWin(): Int

    fun popupOnScoreChange(): Boolean

    fun restoreOnAppReset(): Boolean

    fun vibrateOnTimerFinish(): Boolean

    fun enableDoubleTouch(): Boolean

    fun vibrateOnTimerToggle(): Boolean

    fun pauseOnScoreChange(): Boolean

    fun getBoutLengthMinutes(): Int

}
